src/modules/tools/endstops/Endstops.cpp

```diff
@@ -651,11 +657,47 @@ void Endstops::home(axis_bitmap_t a)
 
     this->axis_to_home= a;
 
+    THEROBOT->disable_segmentation = true; // we must disable segmentation as this won't work with it enabled
+
+    // If "release_first" is enabled, start moving the axes away from the endstops until the switches are released
+    this->status = MOVING_FROM_ENDSTOP_FAST;
+    
+    float delta[homing_axis.size()];
+    for (size_t i = 0; i < homing_axis.size(); ++i) delta[i] = 0;
+    
+    float feed_rate = homing_axis[X_AXIS].fast_rate;
+    bool do_retract = false;
+    for (auto& i : homing_axis) {
+      int c = i.axis_index;
+      if (axis_to_home[c] && i.release_first_enable) {
+        delta[c] = i.max_travel; // we go the max
+        if (!i.home_direction) delta[c] = -delta[c];
+        feed_rate = std::min(i.fast_rate, feed_rate);
+        do_retract = true;
+      }
+    }
+
+    if (do_retract) {
+      THEROBOT->delta_move(delta, feed_rate, homing_axis.size());
+      // wait until finished
+      THECONVEYOR->wait_for_idle();
+
+      // check that the endstops were released 
+      // if the endstop is still not triggered (inverted logic) then enter ALARM state
+      for (auto& i : homing_axis) {
+        int c = i.axis_index;
+        if (axis_to_home[c] && i.release_first_enable && !homing_axis[c].pin_info->triggered) {
+          this->status = NOT_HOMING;
+          THEKERNEL->call_event(ON_HALT, nullptr);
+          THEROBOT->disable_segmentation = false;
+          return;
+        }
+      }
+    }
+
     // Start moving the axes to the origin
     this->status = MOVING_TO_ENDSTOP_FAST;
 
-    THEROBOT->disable_segmentation= true; // we must disable segmentation as this won't work with it enabled
```

Comment:
This is broken I think, as it will not be reset anymore.

Reply:
I don't understand. This is old code.

Reply:
You misunderstood, me sorry. There is no need to set this twice, I was concerned it was not being unset. Seems the diffs are muddled up for some reason and i cannot see the real changes, I'll have to diff the files offline. (Maybe you changed the whitespace).
You can revert the un-needed change, it is starting to get messy.

Reply:
I don't see any diff artefacts from here, including whitespace, in commits before and now:
https://github.com/Smoothieware/Smoothieware/pull/1423/files